src/Compiler/Driver/CompilerDiagnostics.fs

```diff
@@ -735,6 +735,22 @@ let OutputPhasedErrorR (os: StringBuilder) (diagnostic: PhasedDiagnostic) (canSu
                                        | ConstraintSolverError _ as e),
                                        _) -> OutputExceptionR os e
 
+        | ErrorFromAddingTypeEquation (_g, denv, ty1, ty2, ConstraintSolverTupleDiffLengths (_, tl1, tl2, _, _ ), _) ->
+                       
+            let ty1, ty2, tpcs = NicePrint.minimalStringsOfTwoTypes denv ty1 ty2
+            
+            let rec isKnownType = function TType_var ({typar_solution = None}, _) -> false
```

Comment:
* Please don't use `function`
* Please generally avoid nested functions.
* Never match on `TType_*`.  The function you want here is probably `isTyparTy`, but take a look through TypedTreeOps.fsi and get to know those helpers.